package.json

```diff
@@ -141,7 +141,7 @@
     "@types/memoize-one": "^3.1.1",
     "@types/mini-css-extract-plugin": "^0.2.0",
     "@types/mri": "^1.1.0",
-    "@types/node": "^8.10.4",
+    "@types/node": "10.1.4",
```

Comment:
For the moment I'm going to leave this on a `10.1.x` version, despite there being a Node `10.14.9` available as the latest 10.x version. When upgrading to that version I see a number of errors building the app:

```
[WEBPACK] Errors building ask-pass.js
[at-loader] ./node_modules/@types/node/ts3.2/util.d.ts:6:15
    TS2451: Cannot redeclare block-scoped variable 'custom'.
[at-loader] ./node_modules/@types/node/ts3.2/util.d.ts:9:15
    TS2451: Cannot redeclare block-scoped variable 'custom'.
[at-loader] ./app/src/lib/shell.ts:65:64
    TS2345: Argument of type '{ detached: boolean; stdio: (string | WriteStream)[]; }' is not assignable to parameter of type 'SpawnOptions'.
  Types of property 'stdio' are incompatible.
    Type '(string | WriteStream)[]' is not assignable to type '"pipe" | "ignore" | "inherit" | (number | "pipe" | "ipc" | "ignore" | "inherit" | Stream | null | undefined)[] | undefined'.
      Type '(string | WriteStream)[]' is not assignable to type '(number | "pipe" | "ipc" | "ignore" | "inherit" | Stream | null | undefined)[]'.
        Type 'string | WriteStream' is not assignable to type 'number | "pipe" | "ipc" | "ignore" | "inherit" | Stream | null | undefined'.
          Type 'string' is not assignable to type 'number | "pipe" | "ipc" | "ignore" | "inherit" | Stream | null | undefined'.
[at-loader] ./app/src/lib/shell.ts:69:5
    TS2531: Object is possibly 'null'.
[at-loader] ./app/src/lib/shell.ts:74:5
    TS2531: Object is possibly 'null'.
[at-loader] ./app/src/lib/shell.ts:78:5
    TS2531: Object is possibly 'null'.
[at-loader] ./app/src/ui/lib/list/list.tsx:291:30
    TS2345: Argument of type 'number' is not assignable to parameter of type 'Immediate'.
[at-loader] ./app/src/ui/lib/list/list.tsx:294:13
    TS2322: Type 'Immediate' is not assignable to type 'number'.
[at-loader] ./app/src/ui/lib/list/list.tsx:624:22
    TS2345: Argument of type 'number' is not assignable to parameter of type 'Immediate'.
[at-loader] ./app/src/ui/diff/image-diffs/modified-image-diff.tsx:74:28
    TS2345: Argument of type 'number' is not assignable to parameter of type 'Immediate'.
[at-loader] ./app/src/ui/diff/image-diffs/modified-image-diff.tsx:77:11
    TS2322: Type 'Immediate' is not assignable to type 'number'.
[at-loader] ./app/src/ui/history/commit-summary.tsx:145:30
    TS2345: Argument of type 'number' is not assignable to parameter of type 'Immediate'.
[at-loader] ./app/src/ui/history/commit-summary.tsx:148:13
    TS2322: Type 'Immediate' is not assignable to type 'number'.
[at-loader] ./node_modules/@types/node/index.d.ts:82:15
    TS2451: Cannot redeclare block-scoped variable 'custom'.
[at-loader] ./node_modules/@types/node/index.d.ts:85:15
    TS2451: Cannot redeclare block-scoped variable 'custom'.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

These seem to relate some upstream PRs where things have changed in 10.x but the `@types/node` package took time to catch up:

 - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29599
 - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/34285

Given we're not seeing any runtime issues with the app currently, I've avoided updating the contents of `app/` in this PR and have pinned the version for now to focus on the CI changes.

I'd be happy to revisit this in a later situation, such as:

 - we update the app to Electron 5 #6888, which should be tested against Node 12 (and has the correct internal `@types/node` as a dependency)
 - we update `@types/node` manually to a later `10.x` and verify the code that needs to be updated